modules/udf/main.tf

```diff
@@ -14,7 +14,16 @@
  * limitations under the License.
  */
 
-module "bq_find_in_set" {
+ locals {
+   // Regex to extract function name (optionally might include project and dataset ids)
+   // see https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions#temporary-udf-syntax
+   // You can view test cases at: https://regex101.com/r/a7CNA6/2/
+   function_name_regex = "CREATE\\s+(?:OR REPLACE\\s+)?FUNCTION\\s+(?:IF NOT EXISTS)?\\s*((?:[a-zA-Z0-9_]+.)*?(?:[a-zA-Z0-9_]+.)*?[a-zA-Z0-9_]+)(?=\\s*\\()"
+   function_name = regex(local.function_name_regex, var.udf_ddl_query)
+ }
+
+module "udf" {
+  for_each = toset(var.udf_ddl_queries)
```

Comment:
This should be a map of function_name to query.

Reply:
good idea.